File: ngDesk/ngDesk-UI/src/app/models/module-workflow.ts
import { Condition } from './condition';

export class Values {

  constructor(private BODY?: any) { }

  public get body() {
    return this.BODY;
  }

  public set body(body: any) {
    this.BODY = body;
  }

}



export class ConnectionsTo {
  constructor(
    private TITLE: string,
    private FROM: string,
    private TO_NODE: string
  ) { }

  public get title() {
    return this.TITLE;
  }

  public set title(title: string) {
    this.TITLE = title;
  }

  public get from() {
    return this.FROM;
  }

  public set from(from: string) {
    this.FROM = from;
  }

  public get toNode() {
    return this.TO_NODE;
  }

  public set toNode(toNode: string) {
    this.TO_NODE = toNode;
  }
}

export class Plug {
  constructor(
    private ORDER: number,
    private ID: string,
    private NAME: string
  ) { }

  public get order() {
    return this.ORDER;
  }

  public set order(order: number) {
    this.ORDER = order;
  }

  public get id() {
    return this.ID;
  }

  public set id(id: string) {
    this.ID = id;
  }

  public get name() {
    return this.NAME;
  }

  public set name(name: string) {
    this.NAME = name;
  }
}

export class Node {
  constructor(
    private POSITION_X: string,
    private POSITION_Y: string,
    private VALUES: Values,
    private ID: string,
    private TYPE: string,
    private CONNECTIONS_TO: ConnectionsTo[],
    private PLUGS: Plug[],
    private NAME: string
  ) { }

  public get positionX() {
    return this.POSITION_X;
  }

  public set positionX(positionX: string) {
    this.POSITION_X = positionX;
  }

  public get positionY() {
    return this.POSITION_Y;
  }

  public set positionY(positionY: string) {
    this.POSITION_Y = positionY;
  }

  public get values() {
    return this.VALUES;
  }

  public set values(values: Values) {
    this.VALUES = values;
  }

  public get id() {
    return this.ID;
  }

  public set id(id: string) {
    this.ID = id;
  }

  public get type() {
    return this.TYPE;
  }

  public set type(type: string) {
    this.TYPE = type;
  }

  public get connectionsTo() {
    return this.CONNECTIONS_TO;
  }

  public set connectionsTo(connectionsTo: ConnectionsTo[]) {
    this.CONNECTIONS_TO = connectionsTo;
  }

  public get plugs() {
    return this.PLUGS;
  }

  public set plugs(plugs: Plug[]) {
    this.PLUGS = plugs;
  }

  public get name() {
    return this.NAME;
  }

  public set name(name: string) {
    this.NAME = name;
  }
}

export class Workflow {
  constructor(
    private NODES: Node[],
    private LAST_UPDATED_BY?: any,
    private DATE_UPDATED?: any
  ) { }

  public get nodes() {
    return this.NODES;
  }

  public set nodes(nodes: Node[]) {
    this.NODES = nodes;
  }

  public get lastUpdatedBy() {
    return this.LAST_UPDATED_BY;
  }

  public set lastUpdatedBy(lastUpdatedBy: any) {
    this.LAST_UPDATED_BY = lastUpdatedBy;
  }

  public get dateUpdated() {
    return this.DATE_UPDATED;
  }

  public set dateUpdated(dateUpdated: any) {
    this.DATE_UPDATED = dateUpdated;
  }
}

export class ModuleWorkflow {
  constructor(
    private NAME: string,
    private DESCRIPTION: string,
    private CONDITIONS: Condition[],
    private TYPE: {
      DISPLAY: string;
      BACKEND: string;
    },
    private WORKFLOW: Workflow,
    private WORKFLOW_ID?: string,
    private DATE_CREATED?: any,
    private LAST_UPDATED_BY?: any,
    private DATE_UPDATED?: any,
    private ORDER?: any
  ) { }

  public get name() {
    return this.NAME;
  }

  public set name(name: string) {
    this.NAME = name;
  }

  public get description() {
    return this.DESCRIPTION;
  }

  public set description(description: string) {
    this.DESCRIPTION = description;
  }

  public get conditions() {
    return this.CONDITIONS;
  }

  public set conditions(conditions: Condition[]) {
    this.CONDITIONS = conditions;
  }

  public get type() {
    return this.TYPE;
  }

  public set type(type: { DISPLAY: string; BACKEND: string }) {
    this.TYPE = type;
  }

  public get workflow() {
    return this.WORKFLOW;
  }

  public set workflow(workflow: Workflow) {
    this.WORKFLOW = workflow;
  }

  public get workflowId() {
    return this.WORKFLOW_ID;
  }

  public set workflowId(workflowId: string) {
    this.WORKFLOW_ID = workflowId;
  }

  public get dateCreated() {
    return this.DATE_CREATED;
  }

  public set dateCreated(dateCreated: any) {
    this.DATE_CREATED = dateCreated;
  }

  public get lastUpdatedBy() {
    return this.LAST_UPDATED_BY;
  }

  public set lastUpdatedBy(lastUpdatedBy: any) {
    this.LAST_UPDATED_BY = lastUpdatedBy;
  }

  public get dateUpdated() {
    return this.DATE_UPDATED;
  }

  public set dateUpdated(dateUpdated: any) {
    this.DATE_UPDATED = dateUpdated;
  }
  public get order() {
    return this.ORDER;
  }

  public set order(order: any) {
    this.ORDER = order;
  }
}

File: ngDesk-Rest/src/main/java/com/ngdesk/modules/workflows/ModuleWorkflow.java
package com.ngdesk.modules.workflows;

import java.sql.Timestamp;
import java.util.List;

import javax.validation.Valid;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;
import javax.validation.constraints.Size;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.ngdesk.modules.rules.Condition;
import com.ngdesk.workflow.Workflow;

public class ModuleWorkflow {

	@JsonProperty("TYPE")
	@NotNull(message = "WORKFLOW_TYPE_NOT_NULL")
	@Size(min = 1, message = "WORKFLOW_TYPE_NOT_EMPTY")
	@Pattern(regexp = "CREATE|UPDATE|DELETE|CREATE_OR_UPDATE", message = "INVALID_WORKFLOW_TYPE")
	private String type;

	@JsonProperty("NAME")
	@NotNull(message = "WORKFLOW_NAME_NOT_NULL")
	@Size(min = 1, message = "WORKFLOW_NAME_NOT_EMPTY")
	private String name;

	@JsonProperty("DESCRIPTION")
	@NotNull(message = "WORKFLOW_DESCRIPTION_NOT_NULL")
	private String description;

	@JsonProperty("CONDITIONS")
	@NotNull(message = "CONDITIONS_NOT_NULL")
	@Valid
	private List<Condition> conditions;

	@JsonProperty("WORKFLOW_ID")
	private String workflowId;

	@JsonProperty("WORKFLOW")
	@Valid
	private Workflow workflow;

	@JsonProperty("DATE_CREATED")
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
	private Timestamp dateCreated;

	@JsonProperty("DATE_UPDATED")
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
	private Timestamp dateUpdated;

	@JsonProperty("LAST_UPDATED_BY")
	private String lastUpdatedBy;
	
	@JsonProperty("ORDER")
	@NotNull(message = "WORKFLOW_ORDER_NOT_NULL")
	@Min(value = 0, message = "WORKFLOW_ORDER_IS_NULL")
    private int order;

	public ModuleWorkflow() {

	}

	public ModuleWorkflow(
			@NotNull(message = "WORKFLOW_TYPE_NOT_NULL") @Size(min = 1, message = "WORKFLOW_TYPE_NOT_EMPTY") @Pattern(regexp = "CREATE|UPDATE|DELETE|CREATE_OR_UPDATE", message = "INVALID_WORKFLOW_TYPE") String type,
			@NotNull(message = "WORKFLOW_NAME_NOT_NULL") @Size(min = 1, message = "WORKFLOW_NAME_NOT_EMPTY") String name,
			@NotNull(message = "WORKFLOW_DESCRIPTION_NOT_NULL") String description,
			@NotNull(message = "CONDITIONS_NOT_NULL") @Valid List<Condition> conditions, String workflowId,
			@Valid Workflow workflow, Timestamp dateCreated, Timestamp dateUpdated, String lastUpdatedBy,
			@NotNull(message = "WORKFLOW_ORDER_NOT_NULL") @Min(value = 0, message = "WORKFLOW_ORDER_IS_NULL") int order) {
		super();
		this.type = type;
		this.name = name;
		this.description = description;
		this.conditions = conditions;
		this.workflowId = workflowId;
		this.workflow = workflow;
		this.dateCreated = dateCreated;
		this.dateUpdated = dateUpdated;
		this.lastUpdatedBy = lastUpdatedBy;
		this.order = order;
	}

	public String getType() {
		return type;
	}

	public void setType(String type) {
		this.type = type;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public List<Condition> getConditions() {
		return conditions;
	}

	public void setConditions(List<Condition> conditions) {
		this.conditions = conditions;
	}

	public String getWorkflowId() {
		return workflowId;
	}

	public void setWorkflowId(String workflowId) {
		this.workflowId = workflowId;
	}

	public Workflow getWorkflow() {
		return workflow;
	}

	public void setWorkflow(Workflow workflow) {
		this.workflow = workflow;
	}

	public Timestamp getDateCreated() {
		return dateCreated;
	}

	public void setDateCreated(Timestamp dateCreated) {
		this.dateCreated = dateCreated;
	}

	public Timestamp getDateUpdated() {
		return dateUpdated;
	}

	public void setDateUpdated(Timestamp dateUpdated) {
		this.dateUpdated = dateUpdated;
	}

	public String getLastUpdatedBy() {
		return lastUpdatedBy;
	}

	public void setLastUpdatedBy(String lastUpdatedBy) {
		this.lastUpdatedBy = lastUpdatedBy;
	}

	public int getOrder() {
		return order;
	}

	public void setOrder(int order) {
		this.order = order;
	}

	
		

}
